tests/expression_operators/test_null.py

```diff
@@ -44,3 +44,14 @@ def test_is_null_pylist():
     pd_df = pd.DataFrame.from_dict(data)
     pd_df["is_null"] = pd_df["foo"].isnull()
     assert_df_equals(df.to_pandas(), pd_df, sort_key="id")
+
+
+def test_is_null_with_nan_pylist():
```

Comment:
Hey @jaychia, 
I added this test, since it´s not clear to me how "is_null" handles float nan. Based on the documentation is_null should be True for the float nan and the NULL/None value. Based on the test is_null is False for nan but True for None. 
This is not how it´s intended, isn´t it?

Reply:
The intended behavior for `.is_null()` should be:

1. `null` - True
2. `nan` - False
3. Any other value - False

(Pulled from [is_null docs](https://www.getdaft.io/projects/docs/en/stable/api_docs/expression_methods/daft.expressions.Expression.is_null.html#daft.expressions.Expression.is_null)):

```
# [1., None, NaN] -> [False, True, False]
col("x").is_null()
```

---

We use `null` to indicate a "missing value". NaN is a special **float** value that indicates an "invalid number" - "not a number (nan)"!

This is an important distinction to make, especially when working with ML tooling because, for example, neural network models will return NaN on invalid calculations. These values were not missing, they are just invalid float numbers, so as a user you may want to filter them out/handle them separately.

Let me know if I can clarify more on this topic!